game_patch/multi/server.cpp

```diff
@@ -305,6 +316,37 @@ static void send_private_message_with_stats(rf::Player* player)
     send_chat_line_packet(str.c_str(), player);
 }
 
+void shuffle_level_array()
+{
+    std::ranges::shuffle(rf::netgame.levels, rng);
+    xlog::warn("Shuffled level rotation");
+}
+
+const char* get_rand_level_filename()
+{
+    std::size_t num_levels = rf::netgame.levels.size();
+
+    // current_level_index keeps our position in rotation while we go somewhere else
+    rf::String& filename = rf::level_filename_to_load;
+
+    if (num_levels <= 1) {
+        // nowhere else to go, we're staying here!
+        return filename.c_str();
+    }
+
+    std::uniform_int_distribution<std::size_t> dist_levels(0, num_levels - 1);
+    std::size_t rand_level_index = dist_levels(rng);
+
+    // avoid selecting current level
+    if (rf::netgame.levels[rand_level_index] == filename) {
+        rand_level_index = (rand_level_index + 1) % num_levels;
+    }
+
+    filename = rf::netgame.levels[rand_level_index];
```

Comment:
I am rusty with C++. Does this line overwrite `rf::level_filename_to_load`?

Reply:
It does, because I mistakenly only partially reverted [6560196](https://github.com/rafalh/dashfaction/pull/271/commits/656019663371887c368f12bbedcf01170f58fb62)

Fixed now.